certifier-osv.md

```diff
@@ -0,0 +1,197 @@
+- **SBOM Format Support**: 
+  - [CycloneDX](https://cyclonedx.org) (JSON, XML)
+  - [SPDX](https://spdx.dev) (JSON, YAML)
+
+## Integration Details
+
+### Data Collection Process
+
+1. **Dependency Matching**: 
+   - Parses SBOM files for package information
+   - Extracts package names, versions, and ecosystem identifiers
+   - Validates package metadata format
+
+2. **API Querying**: 
+   - Queries OSV using standardized package identifiers
+   - Batch processing for multiple dependencies
+   - Handles API rate limiting and retries
+
+3. **Data Correlation**: 
+   - Maps vulnerabilities to GUAC's dependency graph
+   - Associates vulnerability data with package versions
+   - Maintains relationship between dependencies
+
+### Supported Package Ecosystems
+
+The OSV Certifier supports vulnerability detection across these verified ecosystems:
+
+| Ecosystem | Identifier Format | Example |
+|-----------|------------------|---------|
+| npm | `pkg:npm/{name}@{version}` | `pkg:npm/lodash@4.17.21` |
+| PyPI | `pkg:pypi/{name}@{version}` | `pkg:pypi/requests@2.28.1` |
+| Maven | `pkg:maven/{group}/{artifact}@{version}` | `pkg:maven/org.apache.logging.log4j/log4j-core@2.14.1` |
+| Go | `pkg:golang/{path}@{version}` | `pkg:golang/golang.org/x/text@v0.3.7` |
+| Cargo | `pkg:cargo/{name}@{version}` | `pkg:cargo/serde@1.0.152` |
+| NuGet | `pkg:nuget/{name}@{version}` | `pkg:nuget/Newtonsoft.Json@13.0.1` |
+
+### Feature Support
+
+#### Supported
+- Public vulnerabilities from OSV's database
+- Dependency version mapping with known vulnerabilities
+- Version range analysis
+- [PURL](https://github.com/package-url/purl-spec) package identification
+- Severity classification using [CVSS](https://www.first.org/cvss/) scores
+
+#### Unsupported
+- Private vulnerability detection
+- Ecosystems not covered by OSV
+- Binary vulnerability scanning
+- Custom vulnerability feeds
+
+### Usage
+
+Basic command syntax:
+```bash
+guac osv-certify --sbom <path> [options]
+```
+
+Available options:
+```bash
+Options:
+  --sbom PATH        Path to SBOM file (required)
+  --format FORMAT    SBOM format: cyclonedx|spdx (default: auto-detect)
+  --output PATH     Output file path (default: stdout)
+  --severity LEVEL  Minimum severity: critical|high|medium|low (default: low)
+```
+
+Example usage:
+```bash
+# Basic scan
+guac osv-certify --sbom ./sboms/my_project_sbom.json
+
+# Specify format and severity
+guac osv-certify --sbom ./sbom.xml --format cyclonedx --severity high
+```
+
+## Output Format
+
+### Vulnerability Report Fields
+
+| Field | Description | Example |
+|-------|-------------|---------|
+| id | OSV vulnerability identifier | `OSV-2023-001` |
+| package | Affected package name | `example-library` |
+| version | Affected version | `1.2.3` |
+| severity | Vulnerability severity | `High` |
+| remediation | Fix instructions | `Update to version 1.2.4 or later` |
+
+### Sample Output
+
+```json
```

Comment:
this out put is also wrong, see https://docs.guac.sh/querying-via-cli/#query-to-determine-vulnerabilities for example output

Reply:
```
{
  "_type": "https://in-toto.io/Statement/v0.1",
  "subject": [
    {
      "uri": "pkg:npm/example-library@1.2.3"
    }
  ],
  "predicateType": "https://in-toto.io/attestation/vulns/v0.1",
  "predicate": {
    "scanner": {
      "uri": "osv.dev",
      "version": "0.0.14",
      "result": [
        {
          "id": "GHSA-rc38-5r82-hr3j"
        },
        {
          "id": "CVE-2023-12345"
        }
      ]
    },
    "metadata": {
      "scanStartedOn": "2023-06-06T06:15:28Z",
      "scanFinishedOn": "2023-06-06T06:15:28Z"
    }
  }
}
```